opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -1690,6 +1690,8 @@ assignMassGasRate(data::Wells& wsrpt,
     using rt = data::Rates::opt;
     for (auto& wrpt : wsrpt) {
         auto& well_rates = wrpt.second.rates;
+        if (wrpt.second.dynamicStatus == Opm::Well::Status::SHUT)
+            continue;
```

Comment:
That's fine I guess although it might be better to check `dynamicStatus != OPEN` instead.  Moreover, if we're really going to take action based on the status, then there's no need to intialise the `well_rates` reference before we do the checking.  Another option would be to use [the default value overload](https://github.com/OPM/opm-common/blob/81076046b018d278a50919f6efb3453a6d756a14/opm/output/data/Wells.hpp#L1054-L1058) instead and to write
```C++
const auto w_mass_rate = well_rates.get(rt::gas, 0.0) * gasDensity;
```
In that case we do not check the well status.  I don't immediately know which alternative makes the most sense here.

Reply:
> const auto w_mass_rate = well_rates.get(rt::gas, 0.0) * gasDensity;

I didnt know about this option. Seems like a more robust way of handling this.

Reply:
I have updated the PR accordingly and all my small tests pass. Lets get this in when Jenkins is happy.